pkg/kubelet/cm/event_dispatcher.go

```diff
@@ -41,6 +41,14 @@ type EventDispatcher interface {
 	// PostStopPod is invoked after all of a pod's containers have permanently
 	// stopped running, but before the pod sandbox is destroyed.
 	PostStopPod(cgroupPath string) error
+
+	// PreStartContainer is invoked before the container is created.
+	PreStartContainer(podName, containerName string) (*lifecycle.EventReply, error)
```

Comment:
Could go either way. I'll take a look at how much we'd have to change to send the pod spec in PreStartContainer. If it's too big for the PoC then I feel like it's pretty easy for the isolator to hold onto the copy it receives in PreStartPod and look it up when it receives PreStartContainer.